pyanaconda/modules/subscription/subscription.py

```diff
@@ -302,68 +296,42 @@ def set_system_purpose_data(self, system_purpose_data):
         :param system_purpose_data: system purpose data structure to be set
         :type system_purpose_data: DBus structure
         """
+        # record if the new data is different from current data
+        data_different = self._system_purpose_data != system_purpose_data
```

Comment:
My idea was that the GUI will simply set new SystemPurposeData based on spoke content every time the user exits the spoke - with this check in place, it can do that an avoid triggering call to the `syspurpose` utility and associated log messages polluting the logs.

But thinking about it, it would still have a lot of "System purpose data set" even if nothing changes in the spoke in regards to system purpose, which is not that much better. So yeah, let's drop this check and do it in the GUI, the added code complexity should be not that bad there.